modules/ROOT/pages/testing-oauth-modules.adoc

```diff
@@ -0,0 +1,71 @@
+= Testing OAuth Enabled Connectors
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Mule runtime engine supports authorization through various OAuth grant types including `Authorization Code`. This grant type allows an external system, for example your Mule application, to operate on behalf of a user without the need to share the user's credentials. However, because the `Authorization Code` grant type is designed to force human intervention in the authentication process that identifies users using OAuth (OAuth dance), building an automated test for an application using this grant type is challenging.
+
+You can use the Storage Event processor to bypass the whole OAuth dance phase by manually providing a valid access token. It is your responsibility to manually obtain (and eventually update) such token.
```

Comment:
```suggestion
To overcome this challenge, you can use the Storage Event processor to bypass the OAuth dance by manually providing a valid access token. Note the following requirements:

- It is your responsibility to manually obtain the token.
- It is your responsibility to be aware that the token will eventually expire. 
How often the token expires depends on the service provider to which you are connecting. 
- It is your responsibility to obtain a new token and update the test with it.

```

Reply:
I think we should state the requirements at the very beginning rather than at the end, as caveats. Please consider implementing the previous suggestion (or rewriting to your own liking) and omitting the final text in this file.